execute/tokendata/observer.go

```diff
@@ -203,14 +204,16 @@ func (c *compositeTokenDataObserver) initTokenDataObservations(
 	return tokenObservation
 }
 
+// merge merges token data from two observations, it's used to combine token data from multiple observers.
+// In case of token data mismatch, it returns an error and the base observation.
 func merge(
 	base exectypes.TokenDataObservations,
 	from exectypes.TokenDataObservations,
 ) (exectypes.TokenDataObservations, error) {
 	for chainSelector, chainObservations := range from {
 		for seq, messageTokenData := range chainObservations {
 			if len(messageTokenData.TokenData) != len(base[chainSelector][seq].TokenData) {
-				return nil, errors.New("token data length mismatch")
+				return base, errors.New("token data length mismatch")
```

Comment:
Isn't that an antipattern for Go to return results together with error? How about returning nil here and in parent:

```go
merged, err = merge(tokenDataObservations, tokenData)
if err != nil {
	c.lggr.Error("Error while merging token data", "error", err)
        merged = tokenDataObservations
}
```